dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathResult.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import com.google.cloud.tools.opensource.dependencies.DependencyPath;
+import com.google.cloud.tools.opensource.dependencies.UnresolvableArtifactProblem;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.LinkedListMultimap;
+import java.nio.file.Path;
+
+/** Result of class path resolution with {@link UnresolvableArtifactProblem}s if any. */
+public final class ClassPathResult {
+
+  private final ImmutableList<Path> classPath;
+  private final ImmutableListMultimap<Path, DependencyPath> dependencyPaths;
```

Comment:
I still think we're missing an abstraction here. We have problems of which there are several subtypes including missing artifacts and linkage errors. The linkage errors we don't have a class for. Instead we're storing them as an entry in the multimap.

Reply:
Ping

Reply:
This multimap (`ImmutableListMultimap<Path, DependencyPath> dependencyPaths`) is not `SymbolProblem`s or `ArtifactProblem`s. This is dependency path to Maven artifacts.

For a better representation of linkage errors, I had created an issue https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1083. This PR does not take care of this.

Reply:
#1083 is about the output format. My question is about the data structures we're using in the code itself.

Reply:
Created https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/1139.

This line (line 30) is not linkage errors or artifact problems.